src/service/url-replacements-impl.js

```diff
@@ -549,20 +635,41 @@ export class UrlReplacements {
       if (typeof opt_strargs == 'string') {
         args = opt_strargs.split(',');
       }
-      const binding = (opt_bindings && (name in opt_bindings)) ?
-          opt_bindings[name] : this.getReplacement_(name);
+      let binding;
+      if (opt_bindings && (name in opt_bindings)) {
+        binding = opt_bindings[name];
+      } else if ((binding = this.getReplacement_(name))) {
+        if (opt_sync) {
+          binding = binding.sync;
+          if (!binding) {
+            user().error('ignoring async replacement key: ' + name);
+            return '';
+          }
+        } else {
+          binding = binding.async || binding.sync;
+        }
+      }
       let val;
       try {
         val = (typeof binding == 'function') ?
             binding.apply(null, args) : binding;
       } catch (e) {
         // Report error, but do not disrupt URL replacement. This will
         // interpolate as the empty string.
-        rethrowAsync(e);
+        if (opt_sync) {
+          user().error('Error evaluating replacement: ' + name, e);
+          val = '';
+        } else {
+          rethrowAsync(e);
```

Comment:
What's the difference here between sync and async case? They seem to both report error and that's it. The only diff I see is that first case reports a user error and the second case could be user or dev errors. Is that the idea?